bee-protocol/src/types/milestone_key_manager.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module that provides a milestone key range manager type.
+
+use crate::types::milestone_key_range::MilestoneKeyRange;
+
+use bee_message::milestone::MilestoneIndex;
+
+use std::collections::HashSet;
+
+/// A key manager is managing a set of `MilestoneKeyRange`s.
+#[derive(Clone)]
+pub struct MilestoneKeyManager {
+    min_threshold: usize,
+    key_ranges: Box<[MilestoneKeyRange]>,
```

Comment:
out of interest: I believe there was a discussion around this, but I can't quite remember why we `Box` over a slice here.

Reply:
You'd have to be generic over a lifetime to hold a slice